app/browser/api/ledger.js

```diff
@@ -851,21 +853,28 @@ const shouldTrackTab = (state, tabId) => {
   return !isPrivate && !tabFromState.isEmpty() && ledgerUtil.shouldTrackView(tabFromState)
 }
 
-const addNewLocation = (state, location, tabId = tabState.TAB_ID_NONE, keepInfo = false) => {
+const addNewLocation = (state, location, tabId = tabState.TAB_ID_NONE, keepInfo = false, manualAdd = false) => {
   // We always want to have the latest active tabId
-  const currentTabId = pageDataState.getLastActiveTabId(state)
+  const currentTabId = manualAdd ? tabState.TAB_ID_NONE : pageDataState.getLastActiveTabId(state)
   state = pageDataState.setLastActiveTabId(state, tabId)
-  if (location === currentUrl) {
+  if (location === currentUrl && !manualAdd) {
     return state
   }
 
   // Save previous recorder page
-  if (currentUrl !== locationDefault && currentTabId != null && currentTabId !== tabState.TAB_ID_NONE) {
+  if (currentUrl !== locationDefault && currentTabId != null && currentTabId !== tabState.TAB_ID_NONE && !manualAdd) {
     if (shouldTrackTab(state, currentTabId)) {
       state = addSiteVisit(state, currentTimestamp, currentUrl, currentTabId)
     }
   }
 
+  if (manualAdd) {
+    const minimumVisits = getSetting(settings.PAYMENTS_MINIMUM_VISITS)
+    for (let v = 0; v < minimumVisits; v++) {
+      state = addSiteVisit(state, currentTimestamp, location, currentTabId, manualAdd)
+    }
```

Comment:
I think that we need to return state here, because we don't want to reset `currentUrl` and `currentTimestamp`

Reply:
adjusted

Reply:
adjusted